### PR TITLE
Serialization and more documentation on attributes

### PR DESCRIPTION
**Pull request by Eideren**
There are a couple of spots that are left as todos, some because I don't have a good idea yet as to what they do exactly and others because I didn't have time. But they at least give the reader an idea about some of the other attributes that are there.

It's also a bit dry, I invite anyone willing to build on this PR, fork it if necessary.

### CODE REVIEW

File: en/manual/scripts/serialization.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+public class MyDerivedSerializedClass : MySerializedClass
+{
+    public string MyName;
+}
+```
+
+> [!Note]
+> Your IDE may wrongfully add `System.Runtime.Serialization` to your list of `using` when adding `DataContract`.
+> They are not interchangeable.
+> Make sure that your `DataContract` comes from `Stride.Core`, specifying the namespace explicitly like shown above if necessary.
```
Comment (manio143) [inline en/manual/scripts/serialization.md:28]: nit: it isn't shown above

File: en/manual/scripts/serialization.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+[Stride.Core.DataContract]
+public class MySerializedClass
+{
+    [Stride.Core.DataMember]
+    internal float MyValue;
+}
+```
+
+### [DataAliasAttribute](xref:Stride.Core.DataAliasAttribute)
+Can be used to ensure you do not break previously serialized data whenever you have to change how that member is named in your source.
```
Comment (manio143) [inline en/manual/scripts/serialization.md:45]: I think we need to mention that this only applies for the editor. There is no binary compatibility with  the previous class for compiled assets.

File: en/manual/scripts/serialization.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+## Fields
+
+```cs
+// Read and set in the editor by default
+public object obj;
+
+// Read and set in editor with attribute
+[DataMember] public internal object obj;
+
+// Read only ... readonly are read only.
```
Comment (manio143) [inline en/manual/scripts/serialization.md:87]: I would  note here
```
// Read only fields cannot be modified to point at another object, but the currently set object may be modified
```

File: en/manual/scripts/serialization.md (hunk 2 above)
Comment (Eideren) [reply en/manual/scripts/serialization.md:45]: What do you mean exactly, that the alias is not checked at runtime if they somehow loaded a previous version of the asset hosting that field ?

Comment (manio143) [reply en/manual/scripts/serialization.md:45]: What I mean is that the aliasing is a purely YAML based feature.
In binary serialization the serializer makes an arbitrary decision about field ordering when serializing into a stream of bytes. There is some fingerprinting mechanism which detects the type changed and prevents deserialization from another version of the type.

File: en/manual/scripts/serialization.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
+﻿# Serialization
+
+<span class="label label-doc-level">Beginner</span>
+<span class="label label-doc-audience">Programmer</span>
+
+The editor and serialization system uses four attributes to determine what is serialized and visible in the editor.
```
Comment (IXLLEGACYIXL) [inline en/manual/scripts/serialization.md:6]: There are actually five attributes"DisplayAttribute" which is used to stop the property from showing in the property grid or altering it there but still have it included in yaml and binary serialization

Comment (Eideren) [reply en/manual/scripts/serialization.md:6]: Yep, moved display over into public properties and fields as this is not related to serialization, but more editor styling.

File: en/manual/scripts/serialization.md (hunk 2 above)
Comment (Eideren) [reply en/manual/scripts/serialization.md:45]: Added a note, let me know what you think.

File: en/manual/scripts/serialization.md (hunk 4 above)
Comment (IXLLEGACYIXL) [reply en/manual/scripts/serialization.md:6]: browsable affects the editor serialization.
if its not that important lets leave it out

Comment (Eideren) [reply en/manual/scripts/serialization.md:6]: This is the only read reference I found:
https://github.com/stride3d/stride/blob/master/sources/presentation/Stride.Core.Presentation.Quantum/ViewModels/NodeViewModelFactory.cs#L86-L96
It just hides it from the editor, doesn't affect serialization, unless I am mistaken ?

Comment (IXLLEGACYIXL) [reply en/manual/scripts/serialization.md:6]: allright

File: en/manual/scripts/serialization.md (hunk 2 above)
Comment (manio143) [reply en/manual/scripts/serialization.md:45]: Looks good

**Comment by Eideren** [comment]
@VaclavElias ready when you are :)

**Comment by VaclavElias** [comment]
@Eideren, we need to add the new file `serialisation.md` here 
image

You can add it in /en/manual/toc.md

**Comment by VaclavElias** [comment]
The labels needs to be like so, just different (Bootstrap default) css classes. They removed labels and introduced instead badges..

```
<span class="badge text-bg-primary">Beginner</span>
<span class="badge text-bg-success">Programmer</span>
```

**Comment by Eideren** [comment]
Done :)

**Comment by VaclavElias** [comment]
Last one, regarding Level 2+ headings! 🤣

The titles should be in this order, unless it it s not needed.

```
# Page title - Only one
## Level 2 - Appears in the right menu "In This Article"
### Level 3 
#### And so on..
```

If you would like to have those highlighted in the right menu, you should change them to `##`. Currently they are `###`.

image

Let me know if you would like to update it, otherwise this PR can be merged :)

**Comment by Eideren** [comment]
Nah, I don't want them in the menu, if they aren't there that's fine to me

**Review by VaclavElias — APPROVED** [review]
All seems good :)

**Pull request merged**